oslabs-beta/apollo11
# Networking

**Feature**:
GraphQL network operation detection

**Description**:
- Fix detection logic for `POST` requests to mitigate some gaps
- Add more error checking when attempting to parse requests, such as missing object keys, different data types, etc.
- Relaxed restriction on query parsing, as some sites did not even use the `query` keyword
- Initial attempt at detecting `GET` requests -- likely still somewhat buggy (e.g. generate false positives)
- Added more logging to mitigate missed detections
- Mitigated `eventId` collisions as some network requests have the exact same `startedDateTime` values.  Now storing that as `requestId` and using current `Date()` as `eventId`
- Cleaned up some of the messaging logging

**How to test**:
- Load the extension and navigate to some test websites
- Open the Apollo11 panel and observe the Event Log / Performance tabs
- Inspect the Apollo11 panel for console logs, especially any errors
- To test GraphQL clients that uses `GET` requests, try:
lonelyplanet.com
1800flowers.com
